Sentinel/Classes/EmailSender/EmailSenderTool.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  EmailSenderTool.swift
+//  Sentinel
+//
+//  Created by Antonijo Bezmalinovic on 22.11.2023..
+//
+
+import Foundation
+import MessageUI
+
+@objcMembers
+public final class EmailSenderTool: NSObject, Tool {
+
+    // MARK: - Public properties -
+    
+    public var name: String
+
+    // MARK: - Private properties -
+
+    private let getter: () -> (MailData)
+
+    private var alertTitle: String
+    private var alertMessage: String
+
+    // MARK: - Lifecycle -
+    /// email data  (subject, body, attachnebt...) is provided by 'getter' callback which should be implemented during Sentiel initialization step, for example:
+    /// class SentinelInitializer {
+    /// let configuration = Sentinel.Configuration(
+    ///    trigger: Triggers.shake,
+    ///    sourceScreenProvider: SourceScreenProviders.default,
+    ///    tools: [
+    ///     ...,
+    ///        EmailSenderTool {
+    ///            return DomeProvider.composeMail()
+    ///        },
+    ///     ....,
+    ///     ]
+    /// }
+    /// name that will appear in Sentiel
+    ///
+    /// alertTitle && alertMessage describe alert content that will appear if device is not configured to send email's
+    ///
+    public init(
+        getter: @escaping () -> (MailData),
+        name: String = "Email Sender",
+        alertTitle: String = "Email Not Available",
+        alertMessage: String = "Your device is not configured to send emails. Please set up an email account in Mail app or use another device."
```

Comment:
I understand what documented `init` says, but this can be easily misused.
Still, if you want to give users an opportunity to make custom messages, I would propose an enum:

```swift
enum EmailSenderUnavailableError: Error {
    case `default`
    case custom(title: String, message: String)
}
```

or 

```swift
enum EmailSenderError: Error {
    case unavailable
    case custom(title: String, message: String)
}
```

and then the `init` method would use a default parameter (default or unavailable), which will produce title and message you've defined here.